### PR TITLE
4.4 Identity Management Testing - v4 Migration Cleanup 

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR covers issue #80 .

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix linting issues
- Fix dead links

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 1]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
------------
+## How to Test
 
 In black box testing, the tester knows nothing about the specific application, username, application logic, error messages on log in page, or password recovery facilities. If the application is vulnerable, the tester receives a response message that reveals, directly or indirectly, some information useful for enumerating users.
 
-### HTTP Response message
+### HTTP Response Message
 
-**Testing for Valid user/right password**
+#### Testing for Valid User/Right Password**
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Stars trailing

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 2]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
 
 Record the server answer when you submit a valid user ID and valid password.
 
-**Result Expected:**
+##### Result Expected:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Remove this header. This shouldn't be a section of its own.

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 3]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
 
-**Testing for valid user with wrong password**\
+#### Testing for Valid User with Wrong Password
+
 Now, the tester should try to insert a valid user ID and a wrong password and record the error message generated by the application.
 
-**Result Expected:**\
-The browser should display a message similar to the following one:\
-![](AuthenticationFailed.png "fig:AuthenticationFailed.png")\
+##### Result Expected:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Remove this header.

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 4]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
+
+The browser should display a message similar to the following one:
+![Authentication Failed](images/AuthenticationFailed.png)
+
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Replace the new line with a `\` as discussed in other PRs in order to have less spacing.

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 5]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
+*Figure 4.4.4-1: Authentication Failed*
+
+
+or something like:
+![No Configuration Found](images/NoConfFound.jpg)
 
-or something like:\
-![](NoConfFound.jpg "fig:NoConfFound.jpg")\
+*Figure 4.4.4-2: No Configuration Found*
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Similar comment for the spacing

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 6]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
 
 against any message that reveals the existence of user, for instance, message similar to:\
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Remove the `\` and replace with a new line

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 7]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
 Login for User foo: invalid password
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Set this inside backticks as this is a response message.

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 8]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
-**Result Expected:**
+##### Result Expected:
 
-If the tester enters a nonexistent user ID, they can receive a message similar to:\
-![](Userisnotactive.png "fig:Userisnotactive.png")\
-or message like the following one:\
+If the tester enters a nonexistent user ID, they can receive a message similar to:
+![This User is Not Active](images/Userisnotactive.png)
+
+*Figure 4.4.4-3: This User is Not Active*
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Same comment for images and spacing

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 9]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
+
+or message like the following one:
 
 `Login failed for User foo: invalid Account`
 
 Generally the application should respond with the same error message and length to the different incorrect requests. If the responses are not the same, the tester should investigate and find out the key that creates a difference between the two responses. For example:
 
--   Client request: Valid user/wrong password --&gt; Server answer:'The password is not correct'
--   Client request: Wrong user/wrong password --&gt; Server answer:'User not recognized'
+- Client request: Valid user/wrong password --&gt; Server answer:'The password is not correct'
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Update these 2 lines in a more presentable manner where client requests and server answers are separate. No need to use HTML entities as well.

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 10]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
+#### URI Probing
+
 Sometimes a web server responds differently if it receives a request for an existing directory or not. For instance in some portals every user is associated with a directory. If testers try to access an existing directory they could receive a web server error.
 
-A very common error that is received from web server is:\
-403 Forbidden error code and\
-404 Not found error code\
-Example\
-<http://www.foo.com/account1> - we receive from web server: 403 Forbidden
+A very common error that is received from web server is:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Update these lines as they require new lines in between, or can be done using bullet points, or triple backticks?

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 11]
```diff
@@ -9,116 +10,133 @@ The tester should interact with the authentication mechanism of the application
...
-CN000100\
-CN000101\
-….\
-Sometimes the usernames are created with a REALM alias and then a sequential numbers:\
-R1001 – user 001 for REALM1\
-R2001 – user 001 for REALM2\
-\
+In some cases the user IDs are created with specific policies of administrator or company. For example we can view a user with a user ID created in sequential order:
+
+CN000100
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: This requires new line updates

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 12]
```diff
@@ -127,34 +145,30 @@ Again, we can guess a username from the information received from an LDAP query
...
+Verify that the application answers in the same manner for every client request that produces a failed authentication. For this issue the Black Box testing and Gray Box testing have the same concept based on the analysis of messages or error codes received from web application.
 
-**Testing for Authentication error messages**\
-Verify that the application answers in the same manner for every client request that produces a failed authentication. For this issue the Black Box testing and Gray Box testing have the same concept based on the analysis of messages or error codes received from web application.\
+##### Result Expected:
 
-**Result Expected:**\
-The application should answer in the same manner for every failed attempt of authentication.\
-For Example:\
+The application should answer in the same manner for every failed attempt of authentication.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Doesn't have the expected behavior. Add a new line

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 13]
```diff
@@ -82,7 +82,6 @@ Periodically review the sensitivity of existing design and configuration informa
 3. Bishop Fox&apos;s Google Hacking Diggity Project - http://www.bishopfox.com/resources/tools/google-hacking-diggity/
```
[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: This file supposed to come in 4.2 PR. Can I remove it from this PR?

[the remark this answers is not in the document]
Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Yes, please do.

[the remark this answers is not in the document]
Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Done

File: document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md [hunk 14]
```diff
@@ -82,23 +77,21 @@ Some web application release a specific error code or message that we can analyz
...
 
 #### URI Probing
 
 Sometimes a web server responds differently if it receives a request for an existing directory or not. For instance in some portals every user is associated with a directory. If testers try to access an existing directory they could receive a web server error.
 
-A very common error that is received from web server is:
-403 Forbidden error code and
-404 Not found error code
-Example:
+A very common error that is received from web server is: 403 Forbidden error code and 404 Not found error code
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.4_Identity_Management_Testing/4.4.4_Testing_for_Account_Enumeration_and_Guessable_User_Account_OTG-IDENT-004.md]: Keep them as bullet points. The sentence can be restructured to:
Some of the common errors received from web servers are:

-...
-...

**Comment by rejahrehim** [comment]
@ThunderSon Fixed all based on the comments.

**Pull request merged**